spec/index.bs

```diff
@@ -362,6 +362,9 @@ The precise mechanism of clearing the workgroup memory can differ between platfo
 If the native API does not provide facilities to clear it, the WebGPU implementation transforms the compute
 shader to first do a clear across all invocations, synchronize them, and continue executing developer's code.
 
+The WebGPU implementations should issue a warning about the potential performance penalty when initializing the
+uninitialized resources to zero.
```

Comment:
I think technically this should be non-normative since it's not observable behavior anyway (it's only in the developer console). Additionally, we want all implementations to warn even if there isn't a penalty in that implementation, which means we need to describe roughly the situation in which they should warn.

```suggestion
<div class=note>
    The initialization status of a resource used in a queue operation can only be known when the
    operation is enqueued (not when it is encoded into a command buffer, for example). Therefore,
    some implementations will require an unoptimized late-clear at enqueue time (e.g. clearing a
    texture, rather than changing {{GPULoadOp}} {{GPULoadOp/"load"}} to {{GPULoadOp/"clear"}}).

    In such cases, all implementations **should** issue a developer console warning about the
    potential performance penalty, even if there is no penalty in that implementation.
</div>
```

(Apologies for like, completely rewriting this)

Reply:
@toji please review my suggested text

Reply:
Suggested change LGTM, though I would maybe replace

`In such cases, all implementations **should**`

With something like

`As a result, all implementations **should**`

Since "In such cases" still sounds like it's hinting that it only needs to be done when the performance caveat applies, even though the rest of the sentence contradicts that.

Reply:
@kainino0x it looks much better. Thanks!